pyinv/assets/tests/test_asset_code.py

```diff
@@ -25,3 +31,27 @@ def test_validate_bad_asset_codes(self) -> None:
         for code in self.INVALID_CODES:
             with self.assertRaises(ValidationError):
                 self.strategy.validate(code)
+
+
+class TestStudentRoboticsAssetCodes():
+
+    @pytest.fixture
+    def strategy(self) -> AssetCodeStrategy:
+        return StudentRoboticsAssetCodeStrategy()
+
+    @pytest.mark.parametrize("code", ["sr2UT29", "1VBC"])
+    def test_valid_code(self, code: str, strategy: AssetCodeStrategy) -> None:
+        strategy.validate(code)
+
+    @pytest.mark.parametrize("code", ["srZZZ", "sr~!:@"])
+    def test_invalid_characters(self, code: str, strategy: AssetCodeStrategy) -> None:
+        with pytest.raises(ValidationError, match="Invalid characters in code:"):
+            strategy.validate(code)
+
+    @pytest.mark.parametrize("code", ["sr2UT27", "sr1VBV"])
+    def test_invalid_checkdigit(self, code: str, strategy: AssetCodeStrategy) -> None:
+        with pytest.raises(ValidationError, match="The check digit was invalid."):
+            strategy.validate(code)
+
+    def test_generate(self, strategy: AssetCodeStrategy) -> None:
+        assert strategy.generate_new_code() is None
```

Comment:
Suggestion: I'd be interested in a test which confirms generated codes are valid.

Reply:
We do this for Damm32 code, but as we are not implementing generation of SR code here, this test is sufficient currently.